### ISSUE TITLE
Request: --only-unwatched

### ISSUE THREAD

**Issue by maciekish** [rg3/youtube-dl#14874]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.11.26*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2017.11.26**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

### Description of your *issue*, suggested solution and other information

You already have `--mark-watched` so it would make sense adding `--only-unwatched`. This way it would be possible to keep a "Download"-playlist which automatically downloads each video once and doesn't redownload it once deleted locally.

**Comment by s104346** [comment]
This feature kind of already exists as `--download-archive`:
https://github.com/rg3/youtube-dl#how-do-i-download-only-new-videos-from-a-playlist

**Comment by dstftw** [comment]
`--download-archive`.

**Comment by Joshfindit** [comment]
I'd like to reopen this, especially for playlists.

Two reasons:

1. A video may be added to the Watch Later playlist then watched before `youtube-dl` is run. *Desired outcome: don't download.*
2. A series playlist may have hundreds of videos in a series that have been watched at different times with only a small fraction of them being newly posted (and unwatched). If `youtube-dl` has never been run on this playlist, the only option is to download all videos.

**Comment by Cogito** [comment]
> A series playlist may have hundreds of videos in a series that have been watched at different times with only a small fraction of them being newly posted (and unwatched). If youtube-dl has never been run on this playlist, the only option is to download all videos.

This is my use case as well.

More specifically, I want to keep my downloaded and not downloaded watch history as in-sync as possible. If I've already watched some videos in a playlist I might want to exclude them when being downloaded. If I've downloaded them I want to mark them as watched.

Half of that story is possible now, I was hoping that the other half might be possible using `--match-filter` but haven't found out if so yet. The proposed option would provide the required functionality.
